goosepaper/multiparser.py

```diff
@@ -27,35 +37,150 @@ def __init__(self):
         )
+        self.parser.add_argument(
+            "--noupload",
+            action="store_true",
+            required=False,
+            default=None,
+            help="Overrides any other 'upload: true' arguments from config files or command line. Useful for testing configs or story generation without having to edit config files."
+        )
+        self.parser.add_argument(
+            "--showconfig",
+            action="store_true",
+            required=False,
+            default=None,
+            help="Print out all config files and command line options in order loaded and the final config to help finding conflicting options. Needed since there are now four possible ways to pass options."
+        )
+        self.parser.add_argument(
+            "-n",
+            "--nostory",
+            required=False,
+            default=False,
+            action="store_true",
+            help="||Skip story creation. Combined with \"--upload\" can be used to\nupload a preexisting output file.\n\n** NOTE ** If used without \"--upload\" goosepaper will run but\nperform no action.",
+        )
         self.parser.add_argument(
             "--replace",
             action="store_true",
             required=False,
             default=None,
-            help="Will replace a document with same name in your remarkable.",
+            help="||Will replace a document with same name in your remarkable.\nDefault behaviour is case sensitive (ala *nix/Mac).\n\ne.g. 'A Flock of RSS Feeds.epub' and 'a flock of rss feeds.epub'\nare seen as TWO different files. Can be altered with '--nocase'\nor '--strictlysane' switches.",
         )
         self.parser.add_argument(
-            "-f",
-            "--folder",
+            "--noreplace",
+            action="store_true",
             required=False,
-            help="Folder to which the document will be uploaded in your remarkable.",
+            default=None,
+            help="Only valid when specified on command line (ignored if present in any config file). Supersedes any config file setting for 'replace: true', thus ensuring that the file will NEVER be overwritten. Will also supersede command line '--replace' if both are specified regardless of order."
+            )
+        self.parser.add_argument(
+            "--cleanup",
+            required=False,
+            default=None,
+            action="store_true",
+            help="Delete the output file after upload."
+        )
+        self.parser.add_argument(
+            "--nocase",
+            required=False,
+            default=None,
+            action="store_true",
+            help="** COMING SOON: Enables case insensitivity on upload. By default folders \"My Goosepapers\" and \"my goosepapers\" are NOT the same folder. Enabling this feature means that they would be considered the same. No effect unless \"-u\" is specified."
+        )
+        self.parser.add_argument(
+            "--strictlysane",
```

Comment:
This made me laugh, but it definitely doesn't give me any idea what this flag does as an end-user. Maybe something like `--avoid-filename-collisions`...? I hate to be a funsucker here but my suspicion is that goosepaper cli commands are about to get way more verbose with all these (awesome!) new options, so I think it makes sense to make each option as clear as possible.

Reply:
I think the strictlysane and case options need to be rethought. I'm going to pull them out for now.